### PR TITLE
GH actions update - latest maven, enable native

### PR DESCRIPTION
**Pull request by rsvoboda**
GH actions update - latest maven, enabled native, exclusions updates.

Native mode enabled for Windows and Linux. Windows has limitations due to path length restrictions.

Native part of daily runs were executed in 10+ run (see `.` commits in this PR history, edit: switch from draft to full PR dropped the history of commits ... ) to find all the troubles (exclusions, license agreements, Windows path limits), here is example from the last iteration:
Screenshot 2023-04-28 at 8 37 35

### CODE REVIEW

File: .github/workflows/daily.yml [hunk 1]
```diff
@@ -43,65 +33,34 @@ jobs:
         with:
           name: ci-artifacts
           path: artifacts-linux-jvm${{ matrix.java }}.zip
-  linux-build-released-dev:
```
Comment (jsmrcka) [inline .github/workflows/daily.yml:46]: What is the reason for dropping the dev mode tests?

File: src/main/resources/exclusions.properties [hunk 2]
```diff
@@ -68,6 +69,7 @@ resteasy-jsonb,spring-web=skip-all
 resteasy-multipart,spring-web=skip-all
 resteasy-reactive,oidc-client-filter=skip-all
 spring-web,oidc-client-filter=skip-all
+spring-web,rest-client-jsonb=skip-all
```
Comment (jsmrcka) [inline src/main/resources/exclusions.properties:72]: We should also add
```suggestion
spring-web,rest-client-jsonb=skip-all
spring-web,rest-client-jaxb=skip-all
spring-web,rest-client=skip-all
```
See failures in https://main-jenkins-csb-quarkusqe.apps.ocp-c1.prod.psi.redhat.com/job/quarkus-main-rhel8-jdk11-extensions-combinations-ts/117/.

**Review by jsmrcka — CHANGES_REQUESTED** [review]

File: .github/workflows/daily.yml (hunk 1 above)
Comment (rsvoboda) [reply .github/workflows/daily.yml:46]: they were never enabled

If you think dev mode ones are needed they can be addressed in separate PR

Comment (jsmrcka) [reply .github/workflows/daily.yml:46]: Oh, I see it now, thanks.

**Review by jsmrcka — APPROVED** [review]

**Pull request merged**